runhouse/resources/module.py

```diff
@@ -942,6 +945,94 @@ def _extract_pointers(raw_cls_or_fn: Union[Type, Callable], reqs: List[str]):
 
         return remote_import_path, module_name, cls_or_fn_name
 
+    def generate_openapi_spec(self, spec_name: Optional[str] = None):
+        """Generate an OpenAPI spec for the module.
+
+        TODO: This breaks if the module has type annotations that are classes, and not standard library or
+        typing types.
+
+        Maybe we can do something using: https://github.com/kuimono/openapi-schema-pydantic to allow
+        nested Pydantic models easily as schemas?
+
+        TODO: What happens if there is an empty function, will this work with an empty body even though it is
+        marked as required?
+
+        """
+
+        spec_name = spec_name or self.name
+
+        if not spec_name:
+            raise ValueError(
+                "Module must have a name to generate an OpenAPI spec. Try saving the module first."
+            )
+
+        cls = self.__class__
```

Comment:
I think it's a bad practice to name a variable `cls`, sort like naming a random variable `self`

Reply:
resolved, was a remnant from an earlier version where I was reusing that, killed the variable off now